### PR TITLE
 Move ballerina build, test and publishing to a single task

### PR DESCRIPTION
**Pull request by lahirusamith**
## Purpose
$Subject

issue: https://github.com/ballerina-platform/ballerina-standard-library/issues/1247

## Checklist
- [ ] Linked to an issue

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -82,10 +82,24 @@ Execute the commands below to build from source.
...
+        ./gradlew clean build -PbalJavaDebug=<port>
+        ./gradlew clean test -PbalJavaDebug=<port>
+
+8. Publish ZIP artifact to the local `.m2` repository:
+   
+        ./gradlew clean build publishToMavenLocal
+   
+9. Publish the generated artifacts to the local Ballerina central repository:
+   
+        ./gradlew clean build -PpublishToLocalCentral=true
```
Comment (niveathika) [inline README.md:98]: Github actions also needs to be updated

**Review by niveathika — CHANGES_REQUESTED** [review]

**Comment by niveathika** [comment]
@lahirusamith When this solves an issue only for a specific module do not use fixes keyword. 
This will end up closing the PR.

File: sql-ballerina/build.gradle [hunk 2]
```diff
@@ -151,115 +155,130 @@ task initializeVariables {
...
+                    commandLine 'cmd', '/c', "$balJavaDebugParam $distributionBinPath/bal.bat test ${testCoverageParam} ${groupParams} ${disableGroups} ${debugParams} && exit %%ERRORLEVEL%%"
+                } else {
+                    commandLine 'sh', '-c', "$balJavaDebugParam $distributionBinPath/bal test ${testCoverageParam} ${groupParams} ${disableGroups} ${debugParams}"
+                }
             }
         }
-        // extract bala file to artifact cache directory
-        file("$project.projectDir/target/bala").eachFileMatch(~/.*.bala/) { balaFile ->
+
+        if (needBuildWithTest) {
```
Comment (BuddhiWathsala) [inline sql-ballerina/build.gradle:203]: Should be `else if` please check https://github.com/ballerina-platform/module-ballerina-time/pull/100

File: sql-ballerina/build.gradle [hunk 3]
```diff
@@ -151,115 +155,130 @@ task initializeVariables {
...
+                        commandLine 'cmd', '/c', "$distributionBinPath/bal.bat push && exit %%ERRORLEVEL%%"
+                    } else {
+                        commandLine 'sh', '-c', "$distributionBinPath/bal push"
+                    }
+                }
             } else {
-                commandLine 'sh', '-c', "$distributionBinPath/bal doc"
+                throw new InvalidUserDataException("Central Access Token is Not Present")
             }
         }
```
Comment (BuddhiWathsala) [inline sql-ballerina/build.gradle:263]: Need to have local ballerina push as in here https://github.com/ballerina-platform/module-ballerina-time/pull/100

File: sql-ballerina/build.gradle [hunk 4]
```diff
@@ -151,115 +155,130 @@ task initializeVariables {
...
-}
-
-task ballerinaPublish {
-    doLast {
-        if (project.version.endsWith('-SNAPSHOT')) {
-            return
-        }
-        if (ballerinaCentralAccessToken != null) {
-            println("Publishing to the ballerina central..")
+        if (needPublishToLocalCentral) {
```
Comment (BuddhiWathsala) [inline sql-ballerina/build.gradle:265]: Should be else if.

**Review by BuddhiWathsala — CHANGES_REQUESTED** [review]

File: sql-ballerina/build.gradle (hunk 2 above)
Comment (lahirusamith) [reply sql-ballerina/build.gradle]: fixed in https://github.com/ballerina-platform/module-ballerina-sql/pull/192/commits/1754367aea11f408e3ce4e803bff0117edb52826

File: sql-ballerina/build.gradle (hunk 3 above)
Comment (lahirusamith) [reply sql-ballerina/build.gradle]: check https://github.com/ballerina-platform/module-ballerina-sql/pull/192/commits/1754367aea11f408e3ce4e803bff0117edb52826

File: sql-ballerina/build.gradle (hunk 4 above)
Comment (lahirusamith) [reply sql-ballerina/build.gradle]: fixed in https://github.com/ballerina-platform/module-ballerina-sql/pull/192/commits/1754367aea11f408e3ce4e803bff0117edb52826

**Review by BuddhiWathsala — APPROVED** [review]
LGTM

File: README.md (hunk 1 above)
Comment (lahirusamith) [reply README.md:98]: checked the publish-release.yml file and no changes needed in that file.

**Comment by lahirusamith** [comment]
> @lahirusamith When this solves an issue only for a specific module do not use fixes keyword.
> This will end up closing the PR.

removed the "Fixes" word in the PR message

**Review by niveathika — APPROVED** [review]

**Pull request merged**